src/v/test_utils/archival.h

```diff
@@ -34,12 +34,12 @@ inline storage::disk_log_builder make_log_builder(std::string_view data_path) {
 }
 
 struct segment_spec {
-    size_t start_offset;
-    size_t end_offset;
+    model::offset start_offset;
+    model::offset end_offset;
     size_t size_bytes;
     std::optional<model::timestamp> timestamp{std::nullopt};
-    size_t start_kafka_offset;
-    size_t end_kafka_offset;
+    model::offset start_kafka_offset;
+    model::offset end_kafka_offset;
```

Comment:
these would have different default values. did you check to see if that is a concern?

Reply:
We explicitly initialize these values.